pkg/aws/aws_client/role.go

```diff
@@ -316,4 +316,46 @@ func (client *AWSClient) UntagRole(roleName string, tagKeys []string) error {
 	}
 	_, err := client.IamClient.UntagRole(context.TODO(), input)
 	return err
-}
\ No newline at end of file
+}
+
+func (client *AWSClient) CreateRoleForSharedVPC(roleName, installerRoleArn string, ingressOperatorRoleArn string) (types.Role, error) {
+	statement := map[string]interface{}{
+		"Sid":    "Statement1",
+		"Effect": "Allow",
+		"Principal": map[string]interface{}{
+			"AWS": []string{installerRoleArn, ingressOperatorRoleArn},
+		},
+		"Action": "sts:AssumeRole",
+	}
+
+	assumeRolePolicyDocument, err := completeRolePolicyDocument(statement)
+	if err != nil {
+		fmt.Println("Failed to convert Role Policy Document into JSON: ", err)
+		return types.Role{}, err
+	}
+
+	return client.CreateRole(roleName, string(assumeRolePolicyDocument), "", make(map[string]string), "/")
+}
+
+func (client *AWSClient) CreatePolicyForSharedVPC(policyName string) (string, error) {
```

Comment:
What is this policy used for? Can the statement get from OCM backend? Will the policy be changed in future?

Reply:
This is not resolved. Can you respond?